firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/FirebaseAppDistribution.java

```diff
@@ -132,41 +135,124 @@ public UpdateTask updateIfNewReleaseAvailable() {
 
       cachedUpdateIfNewReleaseTask = new UpdateTaskImpl();
     }
-    checkForNewRelease()
-        .onSuccessTask(
-            release -> {
-              if (release == null) {
-                postProgressToCachedUpdateIfNewReleaseTask(
-                    UpdateProgress.builder()
-                        .setApkFileTotalBytes(UNKNOWN_RELEASE_FILE_SIZE)
-                        .setApkBytesDownloaded(UNKNOWN_RELEASE_FILE_SIZE)
-                        .setUpdateStatus(UpdateStatus.NEW_RELEASE_NOT_AVAILABLE)
-                        .build());
-                setCachedUpdateIfNewReleaseResult();
-                return Tasks.forResult(null);
+
+    if (!isTesterSignedIn()) {
+      Activity currentActivity = lifecycleNotifier.getCurrentActivity();
+      if (currentActivity == null) {
+        LogWrapper.getInstance().e("No foreground activity found.");
+        return getErrorUpdateTask(
+            new FirebaseAppDistributionException(
+                ErrorMessages.APP_BACKGROUNDED,
+                FirebaseAppDistributionException.Status.UPDATE_NOT_AVAILABLE));
+      }
+
+      signInDialog =
+          getSignInDialog(
+              currentActivity,
+              firebaseApp,
+              (dialogInterface, i) ->
+                  checkForNewRelease() // check for new release will call sign in tester
+                      .onSuccessTask(this::handleNewRelease)
+                      .addOnFailureListener(this::onUpdateIfNewReleaseFailure),
+              (dialogInterface, i) -> dismissSignInDialogCallback(),
+              (dialogInterface) -> dismissSignInDialogCallback());
+      signInDialog.show();
+    } else {
+      checkForNewRelease() // check for new release will call sign in tester
+          .onSuccessTask(this::handleNewRelease)
+          .addOnFailureListener(this::onUpdateIfNewReleaseFailure);
+    }
+
+    synchronized (updateIfNewReleaseTaskLock) {
```

Comment:
This synchronized block serves no purpose since it immediately returns anyway.

Reply:
We still need the block, because we are reading the `cachedUpdatedIfNewReleaseTask`, which is guarded by the `updateIfNewReleaseLock` to prevent race conditions.

The alternative would be to remove or ignore the `@GuardedBy` annotation (which I think would be prone to errors)

Reply:
Ah, OK thanks for checking. I'm fine with leaving it for now.